tests/test_mongodb_namespace_restrictions.py

```diff
@@ -5,17 +5,36 @@
 from bson import ObjectId
 
 from mongodb_common import drop_all_collections, get_test_connection, ensure_environment_variables_set
-from tap_tester import connections, menagerie, runner
+from tap_tester import connections, menagerie, runner, LOGGER
 
 
 RECORD_COUNT = {}
 
-# Collection names.  Mongo 4.2 max supported namespace is 120 bytes = 120 characters
+# namespace length = db name + collection name + 1 (for the separator dot)
+#   mongo 4.2 max supported namespace is 120 bytes = 120 characters
+#   mongo 4.4, 5.0, and 6.0 max supported namespace is 255 bytes
+db_name = 'simple_db'
+max_name_length = 120
+
+# collection names
 coll_name_1 = ( "Collection_name_with_120_characters_012345678901234567890123456789"
                 "01234567890123456789012345678901234567890123" )
 coll_name_2 = ( "Collection_name_with_120_characters_123456789012345678901234567890"
                 "12345678901234567890123456789012345678901234" )
-
+coll_name_3 = ( "Collection_name_with_121_characters_234567890123456789012345678901"
+                "234567890123456789012345678901234567890123456")
+coll_name_4 = ( "Collection_name_with_255_characters_345678901234567890123456789012"
+                "345678901234567890123456789012345678901234567890123456789012345678"
+                "901234567890123456789012345678901234567890123456789012345678901234"
+                "56789012345678901234567890123456789012345678901")
+coll_name_5 = ( "Collection_name_with_255_characters_456789012345678901234567890123"
+                "456789012345678901234567890123456789012345678901234567890123456789"
+                "012345678901234567890123456789012345678901234567890123456789012345"
+                "67890123456789012345678901234567890123456789012")
+coll_name_6 = ( "Collection_name_with_256_characters_567890123456789012345678901234"
+                "567890123456789012345678901234567890123456789012345678901234567890"
+                "123456789012345678901234567890123456789012345678901234567890123456"
+                "789012345678901234567890123456789012345678901234")
```

Comment:
Python strings have this funny feature where you can make a string by multiplying a base string by a number

```python
>>> "abc" * 3
'abcabcabc'
>>> "1" * 3
'111'
```

So another way we can do this is to have a little helper function figure out the number of missing characters and then return the full string padded with a character. If we needed random characters, that shouldn't be too hard of a modification either.

```suggestion
def pad_string(starting_string, target_length):
    missing_length = target_length - len(starting_string)
    return starting_string + ("9" * missing_length)

coll_name_3 = pad_string("Collection_name_with_121_characters", 121)
coll_name_4 = pad_string("Collection_name_with_255_characters", 255)
coll_name_5 = pad_string("Collection_name_with_255_characters", 255)
coll_name_6 = pad_string("Collection_name_with_256_characters", 256)
```

Another way if we don't care about the value of the padding character is to just use string formatting

```suggestion
coll_name_3 = f'{"Collection_name_with_121_characters":9<121}'
coll_name_4 = f'{"Collection_name_with_255_characters":9<255}'
coll_name_5 = f'{"Collection_name_with_255_characters":9<255}'
coll_name_6 = f'{"Collection_name_with_256_characters":9<256}'
```

Reply:
Updated